### PR TITLE
More wrapping

### PR DESCRIPTION
**Pull request by erendrake**

### CODE REVIEW

File: InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs [hunk 1]
```diff
@@ -29,10 +29,10 @@ public float MinPosition
 
         public float Position
         {
-            get { return rawServo.Position; }
+            get { return rawServo.Translator.ToExternalPos(rawServo.Position); }
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs]: I need to have access to internal position somehow in GUI on some rare occasions

Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs]: We do? Where?

Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Control/Servo/MechanismBase.cs]: When I add preset I take servo's current position and it assumes internal position. I can change the logic though

File: InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs [hunk 2]
```diff
@@ -20,5 +29,90 @@ public void MoveNext()
...
+            {
+                //because Translator expects position in external coordinates
+                nextPosition = rawServo.Translator.ToExternalPos(nextPosition);
+                rawServo.Translator.Move(nextPosition, rawServo.customSpeed * rawServo.speedTweak);
+            }
+
+            Logger.Log ("[Action] MoveToPreset, index=" + presetIndex + " currentPos = " + rawServo.Position + ", nextPosition=" + nextPosition, Logger.Level.Debug);
+        }
+
+        public void Save(bool propagateSemmetry = false)
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs]: maybe propagateS*y*mmetry?

File: InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs [hunk 3]
```diff
@@ -20,5 +29,90 @@ public void MoveNext()
...
+                }
+            }
+        }
+
+        public void Add(float? position = null)
+        {
+            rawServo.PresetPositions.Add(position == null ? rawServo.Position : position.Value);
+        }
+
+        public void AddHere()
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs]: What is this for?

File: InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs (hunk 2 above)
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs]: Maybe proper spelling :) my bad

File: InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs (hunk 3 above)
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Control/Servo/ServoPreset.cs]: my version of your Add(float) was AddHere, i need to remove it

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs [hunk 4]
```diff
@@ -413,20 +403,8 @@ private void ControlWindow(int windowID)
...
-                                        guiPresetsEnabled = !guiPresetsEnabled;
-                                    else
-                                    {
-                                        associatedServo = servo;
-                                        if (!guiPresetsEnabled)
-                                            guiPresetsEnabled = true;
-                                    }
-                                }
+                                var rowHeight = GUILayout.Height(BUTTON_HEIGHT);
+                                ShowPresets(associatedServo, buttonStyle,rowHeight);
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: It's kinda one-off code, does it really deserve a function?

Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: if we want to keep highlighting and we want to remove RawServo, i kinda needs to be.

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs [hunk 5]
```diff
@@ -970,38 +952,36 @@ private void PresetsEditWindow(int windowID)
             GUILayout.BeginVertical();
 
             GUILayout.BeginHorizontal();
-            GUILayout.Label("Preset position" + (associatedServo.RawServo.Translator.IsAxisInverted ? " (Inv axis)" :""), GUILayout.ExpandWidth(true), rowHeight);
+            GUILayout.Label("Preset position" + (associatedServo.Mechanism.IsAxisInverted ? " (Inv axis)" :""), GUILayout.ExpandWidth(true), rowHeight);
             if (GUILayout.Button("Add", buttonStyle, GUILayout.Width(30), rowHeight))
             {
-                associatedServo.RawServo.PresetPositions.Add(associatedServo.Mechanism.Position);
+                associatedServo.Preset.Add();
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: ah, I see now. Then it seems that we don't really need internal position

File: InfernalRobotics/InfernalRobotics/Module/MuMechToggle.cs [hunk 6]
```diff
@@ -1160,25 +1166,6 @@ public void MotionLockToggle(KSPActionParam param)
 
         public void MoveToPreset(int presetIndex, float delta=0)
         {
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Module/MuMechToggle.cs]: Why did you delete move to preset code?

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs (hunk 4 above)
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: I don't see any connection to highlighting in a code that draws a toggle that opens presets window for selected servo, Am I missing something?

Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: I am just keeping the behavior the same. if we want to stop highlighting that is fine with me. i dont think it would be a bad idea for the config window to highlight your part when you mouse over it.

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs [hunk 7]
```diff
@@ -787,40 +762,28 @@ private void EditorWindow(int windowID)
...
                         servo.Input.Forward = grp.ForwardKey;
 
-                        if (EditorWindowPos.Contains(mousePos))
+                        if (editorWindowPos.Contains(mousePos))
                         {
                             Rect last = GUILayoutUtility.GetLastRect();
                             Vector2 pos = Event.current.mousePosition;
                             bool highlight = last.Contains(pos);
-                            servo.RawServo.part.SetHighlight(highlight, false);
+                            servo.Highlight = highlight;
```
Comment (ZiwKerman) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: Here is the code for highlighting, it has no links to preset window

File: InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs (hunk 4 above)
Comment (erendrake) [inline InfernalRobotics/InfernalRobotics/Gui/ControlsGUI.cs]: opps :P

well we did the same thing twice so i made it a single function.
